### PR TITLE
added endpoint to flag and unflag comment, article and gif

### PR DESCRIPTION
**Pull request by mjmandelah07**

### CODE REVIEW

File: controller/flag-controller
Comment (kelvinator07) [inline controller/flag-controller:1]: This Is Not a JS File, How Were You Able To Test This Code?

File: controller/flag-controller [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+} = require("../utilities/response-utility");
+
+// Function to flag an article by ID
+const flagArticle = async (req, res) => {
+  const { articleId } = req.params;
+  const { flagStatus, flagReason } = req.body;
+
+  // Trim the flagReason property to remove empty spaces or set it to null if flagStatus is false
+  const cleanedFlagReason =
+    flagStatus === false ? null : flagReason ? flagReason.trim() : null;
```
Comment (kelvinator07) [inline controller/flag-controller]: No Need Trimming

Comment (mjmandelah07) [reply controller/flag-controller]: Okay i was just trying to remove white space

File: controller/flag-controller
Comment (mjmandelah07) [reply controller/flag-controller:1]: Will check it, cause it was tested

File: controller/article-controller.js [hunk 2]
```diff
@@ -211,15 +217,7 @@ const getArticleById = async (req, res) => {
...
     const commentRows = commentsResult.rows;
-    const comments = commentRows.map((data) => {
-      return {
-        id: data.id,
-        comment: data.comment,
-        authorId: data.user_id,
-        authorName: data.user_name,
-        createdOn: data.created_on,
-      };
-    });
```
Comment (kelvinator07) [inline controller/article-controller.js]: Put This Back

File: controller/article-controller.js [hunk 3]
```diff
@@ -337,21 +337,15 @@ const getAllArticlesByUserId = async (req, res) => {
...
         // Extract the comments for each article
-        const comments = commentRows.map((data) => {
-          return {
-            id: data.id,
-            comment: data.comment,
-            authorId: data.user_id,
-            authorName: data.user_name,
-            createdOn: data.created_on,
-          };
-        });
```
Comment (kelvinator07) [inline controller/article-controller.js]: Put This Back

File: controller/article-controller.js [hunk 4]
```diff
@@ -467,22 +461,16 @@ const getAllArticles = async (req, res) => {
...
```
Comment (kelvinator07) [inline controller/article-controller.js]: Put This Back

File: controller/gif-controller.js [hunk 5]
```diff
@@ -126,24 +126,22 @@ const getAllGifs = async (req, res) => {
       `;
 
         const commentsResult = await db.query(commentsQuery, [row.id]);
-        const comments = commentsResult.rows.map((commentRow) => ({
-          id: commentRow.id,
-          comment: commentRow.comment,
-          authorId: commentRow.user_id,
-        }));
-        const commentData = {
```
Comment (kelvinator07) [inline controller/gif-controller.js]: Put This Back

**Review by kelvinator07 — APPROVED** [review]

**Pull request merged**